README.md

```diff
@@ -82,66 +76,87 @@ The `samlconf` script may take the following parameters:
+See [4.1 Web Browser SSO Profile](https://www.oasis-open.org/committees/download.php/56782/sstc-saml-profiles-errata-2.0-wd-07.pdf) for more information on the different steps for the SSO profile.
 
-#### samlconf-plugins-api
-This module contains the API that must be implemented for a SAML product in order to run this test kit against that product.
+###### How to implement a plugin jar?
 
-#### implementations
-This module contains implementations of the API for specific SAML products.
+1. Implement a plugin jar for the IdP's authentication implementation. \
+Write a Java or Kotlin class that implements the [IdpSSOResponder](external/samlconf-plugins-api/src/main/java/org/codice/compliance/saml/plugin/IdpSSOResponder.java) interface.
+[DDFIdpSSOResponderProvider](external/implementations/samlconf-ddf-impl/src/main/kotlin/org/codice/compliance/saml/plugin/ddf/DDFIdpSSOResponderProvider.kt) and
+[KeycloakIdpSSOResponderProvider](external/implementations/samlconf-keycloak-impl/src/main/kotlin/org/codice/compliance/saml/plugin/keycloak/KeycloakIdpSSOResponderProvider.kt) can be used as examples.
+1. Package that file into a jar.
+1. Place the jar from step 1 and the IdP's metadata into a directory.
+1. Setup your IdP.
+1. Configure your IdP with the test kit's [SP metadata](deployment/distribution/src/main/resources/samlconf-sp-metadata.xml).
+1. The directory from step 3 should be referred to with the `-i` option when running tests. (See [Run the tests](#run-the-tests))
 
-##### samlconf-ddf-impl
-Plugin and IdP metadata XML for the ddf implementation of IdP. See the [DDF](external/implementations/samlconf-ddf-impl/README.md) in this directory for details.
+#### Formatting
+If during development the build fails due to `format violations`, run `gradlew spotlessApply` to fix the formatting.
 
-##### samlconf-keycloak-impl
-Plugin and IdP metadata XML for the Keycloak implementation of IdP. See the [Keycloak](external/implementations/samlconf-keycloak-impl/README.md) in this directory for details.
-
-### deployment
-This module is the project's full package deployment.
-
-#### distribution
-This module contains all the runtime elements including scripts, jars, and configurations.
+#### Project Structure
+This section will briefly talk about the project structure.
 
-#### docker
-This module contains the logic for building a docker image.
-To build this module you must run the docker task by executing `gradlew build docker`.
+```
+.
+├── ctk - test related modules
+│   │
+│   ├── common - classes relating to utility and verification of the test classes
+│   │               NOTE: Schema verification is run before tests and verifications.
+│   │
+│   └── idp - tests being written against a SAML IdP. The `src` directory of the module is organized by the SAML specification as follows:
+│              * Package: Based on Profile (i.e. WebSSO, Single Logout)
```

Comment:
❓ Does this format correctly?

Reply:
Yes it will look like: 
![screen shot 2018-08-07 at 9 13 32 am](https://user-images.githubusercontent.com/29550062/43788423-3b946628-9a22-11e8-9131-97a5c968b6d7.png)